test/hotspot/jtreg/serviceability/sa/ClhsdbDumpheap.java

```diff
@@ -51,26 +61,133 @@ public static void printStackTraces(String file) {
         }
     }
 
+    private static void verifyDumpFile(File dump, boolean compression) throws Exception {
+        assertTrue(dump.exists() && dump.isFile(), "Could not create dump file " + dump.getAbsolutePath());
+        if (!compression) {
+            printStackTraces(dump.getAbsolutePath());
+            return;
+        } else {
+            String deCompressedFile = "SAdump" + System.currentTimeMillis() + ".hprof";
+            File out = new File(deCompressedFile);
+            try {
+                GZIPInputStream gis = new GZIPInputStream(new FileInputStream(dump));
+                FileOutputStream fos = new FileOutputStream(out);
+                byte[] buffer = new byte[1 << 20];
+                int len = 0;
+                while ((len = gis.read(buffer)) > 0) {
+                    fos.write(buffer, 0, len);
+                }
+            } catch (Exception e) {
+                throw new RuntimeException("Can not decompress the compressed hprof file");
+            }
+            printStackTraces(out.getAbsolutePath());
+            out.delete();
+        }
+    }
+
+    private static class SubTest {
+        private String cmd;
+        private String fileName;
+        private String expectedOutput;
+        boolean compression;
+        boolean needVerify;
+
+        public SubTest(String comm, String fName, String expected, boolean isComp, boolean verify) {
```

Comment:
I think moving the `expected` argument to the last position would make it easier to read all the invocations of `SubTest()`